pkg/detectors/azure_storage/storage.go

```diff
@@ -0,0 +1,190 @@
+	dataStr := string(data)
+
+	// Deduplicate results.
+	names := make(map[string]struct{})
+	for _, matches := range namePat.FindAllStringSubmatch(dataStr, -1) {
+		var name string
+		if matches[1] != "" {
+			name = matches[1]
+		} else {
+			name = matches[2]
+		}
+		if _, ok := testNames[name]; ok {
+			continue
+		}
+		names[name] = struct{}{}
+	}
+	if len(names) == 0 {
+		return results, nil
+	}
+
+	keys := make(map[string]struct{})
+	for _, matches := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		key := matches[1]
+		if _, ok := testKeys[key]; ok {
+			continue
+		}
+		keys[key] = struct{}{}
+	}
+	if len(keys) == 0 {
+		return results, nil
+	}
+
+	// Check results.
+	for name := range names {
+		var s1 detectors.Result
+		for key := range keys {
+			s1 = detectors.Result{
+				DetectorType: s.Type(),
+				Raw:          []byte(key),
+				RawV2:        []byte(fmt.Sprintf(`{"accountName":"%s","accountKey":"%s"}`, name, key)),
+				ExtraData: map[string]string{
+					"Account_name": name,
+				},
+			}
+
+			if verify {
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+
+				isVerified, verificationErr := s.verifyMatch(ctx, client, name, key, s1.ExtraData)
+				s1.Verified = isVerified
+				s1.SetVerificationError(verificationErr, key)
+
+				if s1.Verified {
+					break
+				}
+			}
+			results = append(results, s1)
+		}
+	}
+
+	return results, nil
+}
+
+func (s Scanner) verifyMatch(ctx context.Context, client *http.Client, name string, key string, extraData map[string]string) (bool, error) {
+	// https://learn.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key
+	now := time.Now().UTC().Format(http.TimeFormat)
+	stringToSign := "GET\n\n\n\n\n\n\n\n\n\n\n\nx-ms-date:" + now + "\nx-ms-version:2019-12-12\n/" + name + "/\ncomp:list"
+	accountKeyBytes, _ := base64.StdEncoding.DecodeString(key)
+	h := hmac.New(sha256.New, accountKeyBytes)
+	h.Write([]byte(stringToSign))
+	signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
+
+	url := "https://" + name + ".blob.core.windows.net/?comp=list"
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return false, err
+	}
+
+	req.Header.Set("x-ms-date", now)
+	req.Header.Set("x-ms-version", "2019-12-12")
+	req.Header.Set("Authorization", "SharedKey "+name+":"+signature)
+
+	res, err := client.Do(req)
+	if err != nil {
+		// If the host is not found, we can assume that the accountName is not valid
+		if strings.Contains(err.Error(), "no such host") {
+			return false, nil
+		}
+		return false, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// parse response
+		response := storageResponse{}
+		if err := xml.NewDecoder(res.Body).Decode(&response); err != nil {
+			return false, err
+		}
+
+		// update the extra data with container names only
+		var containerNames []string
```

Comment:
suggestion (optional): We can build the final value directly.
```
var b strings.Builder
for i, c := range response.Containers.Container {
	if i > 0 {
		b.WriteString(", ")
	}
	b.WriteString(c.Name)
}
extraData["container_names"] = b.String()
```

Reply:
Good idea. Done.